bundles/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/se/telegram/Telegram.java

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.vallox.internal.se.telegram;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.core.util.HexUtils;
+import org.openhab.binding.vallox.internal.se.ValloxSEConstants;
+import org.openhab.binding.vallox.internal.se.cache.ValloxExpiringCacheMap;
+import org.openhab.binding.vallox.internal.se.mapper.ChannelDescriptor;
+import org.openhab.binding.vallox.internal.se.mapper.MultipleValueChannel;
+import org.openhab.binding.vallox.internal.se.mapper.TemperatureChannel;
+import org.openhab.binding.vallox.internal.se.mapper.ValloxChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Telegram} class holds telegram state, data and parse methods.
+ *
+ * @author Hauke Fuhrmann - Initial contribution
+ * @author Miika Jukka - Rewrite
+ */
+@NonNullByDefault
+public class Telegram {
+
+    public enum TelegramState {
+
+        // States for received telegram
+        EMPTY("Empty telegram"),
+        NOT_FOR_US("Telegram not for us"),
+        OK("Telegram received OK"),
+        ACK("ACK byte received"),
+        CRC_ERROR("CRC checksum failed"),
+        CORRUPTED("Telegram is corrupted"),
+        NOT_DOMAIN("First byte is not domain byte"),
+        SUSPEND("Stop all traffic"),
+        RESUME("Resume normal use"),
+
+        // States for telegrams to send
+        POLL("Poll"),
+        COMMAND("Command");
+
+        public final String stateDetails;
+
+        private TelegramState(String stateDetails) {
+            this.stateDetails = stateDetails;
+        }
+    }
+
+    private final Logger logger = LoggerFactory.getLogger(Telegram.class);
+    private static Map<String, State> channelsToUpdate = new HashMap<>();
+    public TelegramState state;
+    public byte[] bytes = new byte[6];
```

Comment:
Done. I accomplished it using constructor overloading but I'm not sure why it is less wasteful. The array needs to be created anyway to satisfy NonNullByDefault annotation. And creating shorter array would cause getter methods to fail. Or then I'm missing something obivous.

Reply:
Well you had a constructor that replaced the array entirely at the time, so it would create this default array and then immediately replace it with one provided as a parameter. That is what I meant by wasteful.